src/titanium-data-table.ts

```diff
@@ -0,0 +1,347 @@
+    header {
+      flex-direction: column;
+    }
+  }
+
+  selected-actions {
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+    padding: 16px;
+    color: #1e88e5;
+    font-size: 18px;
+    background-color: #e3f2fd;
+    position: absolute;
+    top: 0px;
+    left: 0px;
+    right: 0px;
+    bottom: 0px;
+  }
+
+  [spacer] {
+    flex: 1 1 auto;
+  }
+
+  table-container {
+    padding-bottom: 12px;
+  }
+
+  table-header {
+    display: flex;
+    flex-direction: row;
+    border-bottom: 1px solid #dadce0;
+  }
+
+  table-header ::slotted(titanium-table-header:last-of-type) {
+    padding-right: 24px;
+  }
+
+  titanium-loading-indicator {
+    align-content: center;
+    justify-content: center;
+    margin: 64px;
+  }
+
+  page-buttons {
+    display: flex;
+    flex-direction: row;
+    justify-content: flex-end;
+    padding: 0 12px 12px 12px;
+  }
+
+  pagination-text {
+    align-self: center;
+    font-size: 12px;
+    font-weight: 400;
+    letter-spacing: 0.011em;
+    line-height: 20px;
+    color: #757575;
+    text-align: right;
+    margin: 0 8px;
+  }
+
+  select-all-checkbox {
+    display: block;
+    flex-shrink: 0;
+    align-self: center;
+    margin: 0 8px 0 24px;
+    width: 22px;
+    height: 22px;
+    cursor: pointer;
+  }
+
+  select-all-checkbox>div {
+    width: 22px;
+    height: 22px;
+    padding: 0;
+  }
+
+  select-all-checkbox svg {
+    fill: #757575;
+  }
+
+  :host([single-select]) select-all-checkbox svg[empty] {
+    fill: #b9b9b9;
+    cursor: not-allowed ;
+  }
+
+  [hidden] {
+    display: none;
+  }`;
+
+  render() {
+    return html
+    `<header>
+  <header-text>${this.title}</header-text>
+  <header-actions>
+    <slot name="table-actions"></slot>
+  </header-actions>
+  <selected-actions hidden>
+    <div>1 selected</div>
+    <div spacer></div>
+    <slot name="selected-actions"></slot>
+  </selected-actions>
+</header>
+<table-container>
+  <table-header>
+    <select-all-checkbox @click="${this._handleSelectAllClick}">
+      <div title="${this.singleSelect ? '' : 'Select all'}" ?hidden="${
+        this.selected.length !== 0}">
+        <svg empty viewBox="0 0 24 24">
+          <path fill="none" d="M0 0h24v24H0V0z" />
+          <path d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z" />
+        </svg>
+      </div>
+      <div title="Deselect all" ?hidden="${
+        this.selected.length === 0 ||
+        this.selected.length !== this.items.length}">
+        <svg checked viewBox="0 0 24 24">
+          <path d="M0 0h24v24H0z" fill="none" />
+          <path d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z" />
+        </svg>
+      </div>
+      <div title="Clear selection" ?hidden="${
+        this.selected.length === 0 ||
+        this.selected.length === this.items.length}">
+        <svg indetermanite viewBox="0 0 24 24">
```

Comment:
Does indetermanite do anything? It is spelled wrong

Reply:
It doesn't, removed.